_studio/mfx_lib/scheduler/include/mfx_scheduler_core_thread.h

```diff
@@ -41,8 +41,9 @@ struct MFX_SCHEDULER_THREAD_CONTEXT
     {}
 
     enum State {
-        Waiting, // thread is waiting for incoming tasks
-        Running  // thread is executing a task
+        Waiting,  // thread is waiting for incoming tasks
+        Running,  // thread is executing a task
+        Stopping, // thread is going to stop
```

Comment:
IMO, [Stopping] is quite weird state for thread. If [bQuit] is really unacceptable there is yet another common trick - submit special 'stop task' that will signal a thread to stop.

Reply:
I know. And I feel your pain. Right now in current scheduler implementation quality that's the best I can suggest. Right now we don't have a reasonable way to send a special task.